Makefile

```diff
@@ -45,18 +45,38 @@ lint:
 	@golangci-lint run
 .PHONY: lint
 
-## test: Running all *_test.go.
+## test-unit: Running unit tests
+test-unit:
+	@echo "--> Running unit tests"
+	@go test -v `go list ./... | grep -v node/tests`
+.PHONY: test-unit
+
+## test-unit-race: Running unit tests with data race detector
+test-unit-race:
+	@echo "--> Running unit tests with data race detector"
+	@go test -v -race `go list ./... | grep -v node/tests`
+.PHONY: test-unit-race
+
+## test-swamp: Running swamp tests located in node/tests
+test-swamp:
+	@echo "--> Running swamp tests"
+	@go test -v ./node/tests/swamp
+.PHONY: test-swamp
+
+## test-swamp: Running swamp tests with data race detector located in node/tests
+test-swamp-race:
+	@echo "--> Running swamp tests with data race detector"
+	@go test -v -race ./node/tests/swamp
+.PHONY: test-swamp-race
+
+## test-all: Running both unit and swamp tests
 test:
-	@echo "--> Running tests"
-	@go test -v ./...
+	@echo "--> Running all tests without data race detector"
+	@go test ./...
+	@echo "--> Running all tests with data race detector"
+	@go test -race ./...
```

Comment:
Wanted to say no to this, but later realized it might be useful to understand/see the diff that tests are failing due to races only.

Reply:
So this is automatically resolved.

Reply:
Yeap. It is indeed useful for rn. Regarding execution time - it's relatively fast, so we can consider rethinking this part later on as tests start to grow in numbers.